azurelinuxagent/common/osutil/default.py

```diff
@@ -698,14 +698,17 @@ def _get_all_interfaces(self):
 
         buff = array.array('B', b'\0' * array_size)
         param = struct.pack('iL', array_size, buff.buffer_info()[0])
+
         sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
         ret = fcntl.ioctl(sock.fileno(), IOCTL_SIOCGIFCONF, param)
         retsize = (struct.unpack('iL', ret)[0])
+        sock.close()
+
         if retsize == array_size:
             logger.warn(('SIOCGIFCONF returned more than {0} up '
                          'network interfaces.'), expected)
-        sock = None
-        ifconf_buff = buff.tostring()
+
+        ifconf_buff = buff.tobytes()
```

Comment:
The tool complained that tostring() was deprecated, so I changed it to tobytes() per the recommendation.

Reply:
Actually, I was wrong.  There appears to be another issue lurking here, so I am reverting it back.